libs/banglejs/jswrap_bangle.c

```diff
@@ -5073,27 +5073,38 @@ JsVar *jswrap_banglejs_getPressure() {
   /* Occasionally on some devices (https://github.com/espruino/Espruino/issues/2137)
   you can get an I2C error. This stops the error from being fired when getPressure
   is called and instead rejects the promise. */
+  JsVar *exception = jspGetException();
+  if (exception) {
+    jspromise_reject(promisePressure, exception);
+    jsvUnLock2(promisePressure, exception);
+    JsVar *r = promisePressure;
+    promisePressure = 0;
+    return r;
+  }
+
   bool hadError = jspHasError();
   if (hadError) {
-    JsVar *exception = jspGetException();
+    JsVar *msg = jsvNewFromString("I2C barometer error");
+    JsVar *exception = jswrap_internalerror_constructor(msg);
```

Comment:
I believe I have the locking correct here - `jswrap_internalerror_constructor` unlocks `msg`, but would appreciate a second pair of eyes

Reply:
I don't think `jswrap_internalerror_constructor` does unlock msg - if it's a function called from JS (which I'm pretty sure it is as it's the constructor) then the caller is responsible for unlocking

Reply:
I might've misread the code - I think [here it unlocks](https://github.com/espruino/Espruino/blob/deedeafd91683359807fe77aa2bfb2b37d56d219/src/jswrap_error.c#L57-L57) in `jsvObjectSetChildAndUnLock`

https://github.com/espruino/Espruino/blob/deedeafd91683359807fe77aa2bfb2b37d56d219/src/jswrap_error.c#L51-L62

(⬆️  `msg` here)

https://github.com/espruino/Espruino/blob/deedeafd91683359807fe77aa2bfb2b37d56d219/src/jsvar.c#L3195-L3198

(⬆️  `child` here)

Reply:
Ahh - `jsvObjectSetChildAndUnLock(d, "message", msg); ` does unlock `msg` but on the line above, `msg = jsvAsString(msg);` replaces `msg` with a newly locked version of `msg` that's guaranteed to be a string. I guess for readability it should really be written to a new variable instead.

Reply:
That makes sense - I could make that change as part of this PR if you like?

Reply:
No, it's ok - i'll do it separately now...

Reply:
but you still need to unlock `msg`...

Reply:
Ah yes - I was assuming `jsvAsString` unlocked the value it was given, with us replacing it. Thanks for the pointer!